include/engine/api/nearest_api.hpp

```diff
@@ -38,8 +38,50 @@ class NearestAPI final : public BaseAPI
                        phantom_nodes.front().end(),
                        waypoints.values.begin(),
                        [this](const PhantomNodeWithDistance &phantom_with_distance) {
-                           auto waypoint = MakeWaypoint(phantom_with_distance.phantom_node);
+                           auto &phantom_node = phantom_with_distance.phantom_node;
+                           auto waypoint = MakeWaypoint(phantom_node);
                            waypoint.values["distance"] = phantom_with_distance.distance;
+
+                           util::json::Array nodes;
+
+                           std::uint64_t from_node = static_cast<std::uint64_t>(SPECIAL_OSM_NODEID);
+                           std::uint64_t to_node = static_cast<std::uint64_t>(SPECIAL_OSM_NODEID);
+
+                           auto segment_id = phantom_node.forward_segment_id.id;
+                           if (segment_id != SPECIAL_SEGMENTID)
+                           {
+                               const auto geometry_id = facade.GetGeometryIndex(segment_id).id;
+                               std::vector<NodeID> geometry =
+                                   facade.GetUncompressedForwardGeometry(geometry_id);
+
+                               auto osm_node_id = facade.GetOSMNodeIDOfNode(
+                                   geometry[phantom_node.fwd_segment_position]);
+                               to_node = static_cast<std::uint64_t>(osm_node_id);
+                           }
+
+                           segment_id = phantom_node.reverse_segment_id.id;
+                           if (segment_id != SPECIAL_SEGMENTID)
+                           {
+                               const auto geometry_id = facade.GetGeometryIndex(segment_id).id;
+                               std::vector<NodeID> geometry =
+                                   facade.GetUncompressedForwardGeometry(geometry_id);
+                               auto osm_node_id = facade.GetOSMNodeIDOfNode(
+                                   geometry[phantom_node.fwd_segment_position + 1]);
+                               from_node = static_cast<std::uint64_t>(osm_node_id);
+                           }
+
+                           if (!phantom_node.forward_segment_id.enabled)
+                           {
+                               nodes.values.push_back(to_node);
+                               nodes.values.push_back(from_node);
+                           }
+                           else
+                           {
+                               nodes.values.push_back(from_node);
+                               nodes.values.push_back(to_node);
+                           }
+                           waypoint.values["nodes"] = std::move(nodes);
```

Comment:
The current implementation may return `SPECIAL_OSM_NODEID` . This happens while doing a scan of Monaco area.
Would you suggest using an another special osm id value? or not returning such value in the response?